test/specs/data_sources/model/TraitDataVariable.ts

```diff
@@ -0,0 +1,361 @@
+        id: 'test-image-datasource',
+        records: [{ id: 'id1', value: 'url-to-cat-image' }],
+      };
+      dsm.add(inputDataSource);
+
+      const cmp = cmpRoot.append({
+        type: 'image',
+        tagName: 'img',
+        traits: [
+          {
+            type: 'text',
+            name: 'src',
+            value: {
+              type: 'data-variable',
+              defaultValue: 'default',
+              path: `${inputDataSource.id}.id1.value`,
+            },
+          },
+        ],
+      })[0];
+
+      const img = cmp.getEl() as HTMLImageElement;
+      expect(img?.getAttribute('src')).toBe('url-to-cat-image');
+      expect(cmp?.getAttributes().src).toBe('url-to-cat-image');
+
+      const testDs = dsm.get(inputDataSource.id);
+      testDs.getRecord('id1')?.set({ value: 'url-to-dog-image' });
+
+      expect(img?.getAttribute('src')).toBe('url-to-dog-image');
+      expect(cmp?.getAttributes().src).toBe('url-to-dog-image');
+    });
+  });
+
+  describe('link component', () => {
+    test('component initializes and updates data-variable value', () => {
+      const inputDataSource: DataSourceProps = {
+        id: 'test-link-datasource',
+        records: [{ id: 'id1', value: 'url-to-cat-image' }],
+      };
+      dsm.add(inputDataSource);
+
+      const cmp = cmpRoot.append({
+        type: 'link',
+        tagName: 'a',
+        traits: [
+          {
+            type: 'text',
+            name: 'href',
+            value: {
+              type: 'data-variable',
+              defaultValue: 'default',
+              path: `${inputDataSource.id}.id1.value`,
+            },
+          },
+        ],
+        components: [{ tagName: 'span', content: 'Link' }],
+      })[0];
+
+      const link = cmp.getEl() as HTMLLinkElement;
+      expect(link?.href).toBe('http://localhost/url-to-cat-image');
+
+      const testDs = dsm.get(inputDataSource.id);
+      testDs.getRecord('id1')?.set({ value: 'url-to-dog-image' });
+
+      expect(link?.href).toBe('http://localhost/url-to-dog-image');
+      expect(cmp?.getAttributes().href).toBe('url-to-dog-image');
+    });
+  });
+
+  describe('changeProp', () => {
+    test('component initializes and updates data-variable value using changeProp', () => {
+      const inputDataSource: DataSourceProps = {
+        id: 'test-change-prop-datasource',
+        records: [{ id: 'id1', value: 'I love grapes' }],
+      };
+      dsm.add(inputDataSource);
+
+      const cmp = cmpRoot.append({
+        tagName: 'div',
+        type: 'default',
+        traits: [
+          {
+            name: 'test-change-prop',
+            type: 'text',
+            changeProp: true,
+            value: {
+              type: DataVariableType,
+              defaultValue: 'default',
+              path: `${inputDataSource.id}.id1.value`,
+            },
+          },
+        ],
+      })[0];
+
+      let property = cmp.get('test-change-prop');
+      expect(property).toBe('I love grapes');
```

Comment:
Without said change how might this initial value look? Is it expected to have an initial value with `changeProp`? @artf

Reply:
mmm yeah, now that I'm looking at it and I think you're right... we should get the value from the trait if it's defined, but let's handle the case where the value of the trait is not defined.

```js
{
        tagName: 'div',
        type: 'default',
        'test-change-prop': 'initValue',
        traits: [
          {
            name: 'test-change-prop',
            type: 'text',
            changeProp: true,
          },
        ],
      }
```

Reply:
* https://github.com/GrapesJS/grapesjs/pull/6101